src/Umbraco.Core/PropertyEditors/DataValueReferenceCollection.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+using Umbraco.Core.Composing;
+
+namespace Umbraco.Core.PropertyEditors
+{
+    public class DataValueReferenceCollection : BuilderCollectionBase<IDataValueReference>
```

Comment:
Like i mentioned in the spec and on Slack - this collection does not track `IDataValueReference`, we need a new `IDataReferenceValueFactory` (or whatever you want to call it). `IDataValueReference` is an interface that can be imlemented directly by property editors, `IDataValueReferenceFactory` is something that can create an instance of `IDataValueReference`. 

Prob best to re-read the entire spec again :P